snuba/admin/static/rpc_endpoints/index.tsx

```diff
@@ -1,240 +1,304 @@
-                        >
-                            Copy to Request Body
-                        </Button>
-                    </Accordion.Panel>
-                </Accordion.Item>
-            </Accordion>
-            <Space h="md" />
-            <Textarea
-                label="Request Body (JSON)"
-                placeholder="Enter request body"
-                value={requestBody}
-                onChange={(event) => setRequestBody(event.currentTarget.value)}
-                style={{ width: '100%' }}
-                autosize
-                minRows={5}
-            />
-            <Space h="md" />
-            <Checkbox
-                label="Enable Debug Mode"
-                checked={debugMode}
-                onChange={(event) => setDebugMode(event.currentTarget.checked)}
-                disabled={!DEBUG_SUPPORTED_VERSIONS.includes(selectedVersion || '')}
-                className={classes.debugCheckbox}
-            />
+  return (
+    <div>
+      <h2>RPC Endpoints</h2>
+      <Select
+        label="Select an endpoint"
+        placeholder="Choose an endpoint"
+        data={endpoints.map(endpoint => ({ value: endpoint.name, label: `${endpoint.name} (${endpoint.version})` }))}
+        value={selectedEndpoint}
+        onChange={handleEndpointSelect}
+        style={{ width: '100%', marginBottom: '1rem' }}
+      />
+      <Space h="md" />
+      <Accordion
+        classNames={{ item: classes.accordion }}
+        variant="filled"
+        radius="sm"
+        value={accordionOpened ? 'example' : null}
+        onChange={(value) => setAccordionOpened(value === 'example')}
+      >
+        <Accordion.Item value="example">
+          <Accordion.Control>Example Request Payload</Accordion.Control>
+          <Accordion.Panel>
+            <Code block style={{ color: 'green' }}>
+              <pre>
+                {JSON.stringify(
+                  selectedEndpoint && selectedVersion
+                    ? exampleRequestTemplates[selectedEndpoint]?.[selectedVersion] || exampleRequestTemplates.default
+                    : exampleRequestTemplates.default,
+                  null,
+                  2
+                )}
+              </pre>
+            </Code>
             <Button
-                onClick={handleExecute}
-                disabled={!selectedEndpoint || !requestBody || isLoading}
-                leftIcon={isLoading ? <Loader size="xs" /> : null}
+              onClick={() => {
+                setRequestBody(
+                  JSON.stringify(
+                    selectedEndpoint && selectedVersion
+                      ? exampleRequestTemplates[selectedEndpoint]?.[selectedVersion] || exampleRequestTemplates.default
+                      : exampleRequestTemplates.default,
+                    null,
+                    2
+                  )
+                );
+                setAccordionOpened(false);
+              }}
+              style={{ marginTop: '1rem' }}
             >
```

Comment:
Seems like this is encapsulating the example requests in an accordion, can you make this its own function?